infrastructure/src/main/java/org/corfudb/infrastructure/log/SegmentHandle.java

```diff
@@ -0,0 +1,75 @@
+package org.corfudb.infrastructure.log;
+
+import lombok.Data;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import java.nio.channels.FileChannel;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * The global log is partition into segments, each segment contains a range of consecutive
+ * addresses. Accessing the address space for a particular segment happens through this class.
+ *
+ * @author Maithem
+ */
+@Slf4j
+@Data
+class SegmentHandle {
+    final long segment;
+
+    @NonNull
+    final FileChannel writeChannel;
+
+    @NonNull
+    final FileChannel readChannel;
+
+    @NonNull
+    final FileChannel trimmedChannel;
+
+    @NonNull
+    final FileChannel pendingTrimChannel;
+
+    @NonNull
+    String fileName;
+
+    private Map<Long, AddressMetaData> knownAddresses = new ConcurrentHashMap();
+    private Set<Long> trimmedAddresses = Collections.newSetFromMap(new ConcurrentHashMap<>());
+    private Set<Long> pendingTrims = Collections.newSetFromMap(new ConcurrentHashMap<>());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the "pendingTrims" field and declare it as a local variable in the relevant methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1450)